File: Compose/YABA/app/src/main/java/dev/subfly/yaba/core/state/creation/mention/MentionCreationEvent.kt
package dev.subfly.yaba.core.state.creation.mention

sealed interface MentionCreationEvent {
    data class OnInit(
        val initialText: String = "",
        val initialBookmarkId: String? = null,
        val isEdit: Boolean = false,
        val editPos: Int? = null,
    ) : MentionCreationEvent

    data class OnChangeMentionText(
        val text: String,
    ) : MentionCreationEvent

    /**
     * User finished the bookmark picker; [bookmarkId] is the only payload from selection.
     * Mention text is auto-filled from the bookmark label when it is still blank.
     */
    data class OnBookmarkPickedFromSelection(
        val bookmarkId: String,
    ) : MentionCreationEvent
}

File: Darwin/YABACore/State/Creation/Mention/MentionCreationEvent.swift
//
//  MentionCreationEvent.swift
//  YABACore
//
//  Parity with Compose `MentionCreationEvent` (creation/mention).
//

import Foundation

public enum MentionCreationEvent: Sendable {
    case onInit(
        initialText: String,
        initialBookmarkId: String?,
        isEdit: Bool,
        editPos: Int?
    )
    case onChangeMentionText(String)
    /// User finished bookmark selection; host may supply `bookmarkLabel` so mention text can auto-fill when still blank.
    case onBookmarkPickedFromSelection(bookmarkId: String, bookmarkLabel: String?)
}
